Applications/Browser/BrowserConsoleClient.cpp

```diff
@@ -0,0 +1,123 @@
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "BrowserConsoleClient.h"
+#include "ConsoleWidget.h"
+#include <AK/StringBuilder.h>
+#include <LibGUI/BoxLayout.h>
+#include <LibGUI/JSSyntaxHighlighter.h>
+#include <LibGUI/TextBox.h>
+#include <LibJS/Interpreter.h>
+#include <LibWeb/DOM/DocumentType.h>
+#include <LibWeb/DOM/ElementFactory.h>
+#include <LibWeb/DOM/HTMLBodyElement.h>
+#include <LibWeb/DOM/Text.h>
+#include <LibWeb/DOMTreeModel.h>
+
+namespace Browser {
+
+JS::Value BrowserConsoleClient::log()
+{
+    m_console_widget.print_html(interpreter().join_arguments());
+    return JS::js_undefined();
+}
+
+JS::Value BrowserConsoleClient::info()
+{
+    m_console_widget.print_html(String::format("(i) %s", interpreter().join_arguments().characters()));
+    return JS::js_undefined();
+}
+
+JS::Value BrowserConsoleClient::debug()
+{
+    StringBuilder html;
+    html.append("<i>");
+    html.append(interpreter().join_arguments());
+    html.append("</i>");
+    m_console_widget.print_html(html.string_view());
+    return JS::js_undefined();
+}
+
+JS::Value BrowserConsoleClient::warn()
+{
+    StringBuilder html;
+    html.append("<u>");
+    html.append(interpreter().join_arguments());
+    html.append("</u>");
+    m_console_widget.print_html(html.string_view());
+    return JS::js_undefined();
+}
+
+JS::Value BrowserConsoleClient::error()
+{
+    StringBuilder html;
+    html.append("<b>");
+    html.append(interpreter().join_arguments());
+    html.append("</b>");
+    m_console_widget.print_html(html.string_view());
+    return JS::js_undefined();
+}
+
+JS::Value BrowserConsoleClient::clear()
+{
+    m_console_widget.clear_output();
+    return JS::js_undefined();
+}
+
+JS::Value BrowserConsoleClient::trace()
+{
+    StringBuilder html;
+    html.append(interpreter().join_arguments());
+    auto trace = interpreter().get_trace();
+    for (auto& function_name : trace) {
+        if (function_name.is_empty())
+            function_name = "<anonymous>";
+        html.appendf(" -> %s\n", function_name.characters());
```

Comment:
should that be

```suggestion
        html.appendf(" -> %s<br>", function_name.characters());
```

?

Reply:
Yep absolutely, must have missed it in the conversion from `ReplConsoleClient`.